codalab-service.py

```diff
@@ -0,0 +1,387 @@
+        'uid': None,
+        'service_home': None,
+        'mysql_mount': None,
+        'worker_dir': None,
+        'bundle_stores': [],
+        'worker_docker_network_name': 'codalab_worker_network',
+        'http_port': '80',
+        'rest_port': None,
+        'frontend_port': None,
+        'mysql_port': None,
+        'use_ssl': False,
+        'ssl_cert_file': None,
+        'ssl_key_file': None,
+        'follow': False,
+        'tail': None,
+        'tests': ['default'],
+    }
+
+    ARG_TO_ENV_VAR = {
+        'version': 'CODALAB_VERSION',
+        'dev': 'CODALAB_DEV',
+        'push': 'CODALAB_PUSH',
+        'docker_user': 'DOCKER_USER',
+        'docker_pwd': 'DOCKER_PWD',
+        'user_compose_file': 'CODALAB_USER_COMPOSE_FILE',
+        'start_worker': 'CODALAB_START_WORKER',
+        'initial_config': 'CODALAB_INITIAL_CONFIG',
+        'mysql_root_pwd': 'CODALAB_MYSQL_ROOT_PWD',
+        'mysql_user': 'CODALAB_MYSQL_USER',
+        'mysql_pwd': 'CODALAB_MYSQL_PWD',
+        'root_user': 'CODALAB_ROOT_USER',
+        'root_pwd': 'CODALAB_ROOT_PWD',
+        'uid': 'CODALAB_UID',
+        'service_home': 'CODALAB_SERVICE_HOME',
+        'mysql_mount': 'CODALAB_MYSQL_MOUNT',
+        'worker_dir': 'CODALAB_WORKER_DIR',
+        'worker_docker_network_name': 'CODALAB_WORKER_DOCKER_NETWORK_NAME',
+        'http_port': 'CODALAB_HTTP_PORT',
+        'rest_port': 'CODALAB_REST_PORT',
+        'frontend_port': 'CODALAB_FRONTEND_PORT',
+        'mysql_port': 'CODALAB_MYSQL_PORT',
+        'ssl_cert_file': 'CODALAB_SSL_CERT_FILE',
+        'ssl_key_file': 'CODALAB_SSL_KEY_FILE',
+    }
+
+    @staticmethod
+    def _get_parser():
+        parser = argparse.ArgumentParser(description="Manages your local CodaLab worksheets back-end deployment using docker-compose")
+        subparsers = parser.add_subparsers(dest='command', description='Command to run')
+
+        # SUBCOMMANDS
+
+        start_cmd = subparsers.add_parser('start', help='Start a CodaLab backend instance')
+        logs_cmd = subparsers.add_parser('logs', help='View logs for existing CodaLab instance')
+        test_cmd = subparsers.add_parser('test', help='Run tests against an existing CodaLab instance')
+        build_cmd = subparsers.add_parser('build', help='Build CodaLab docker images using the local codebase')
+        run_cmd = subparsers.add_parser('run', help='Run a command inside a service container')
+
+        stop_cmd = subparsers.add_parser('stop', help='Stop any existing CodaLab backend instance')
+        down_cmd = subparsers.add_parser('down', help='Bring down any existing CodaLab backend instance')
+        restart_cmd = subparsers.add_parser('restart', help='Restart any existing CodaLab backend instance')
+
+        #  BUILD SETTINGS
+
+        for cmd in [build_cmd, start_cmd]:
+            cmd.add_argument('--version', '-v', type=str, help='CodaLab version to use for building and deployment', default=argparse.SUPPRESS)
```

Comment:
why do you want to suppress these?

Reply:
This is a trick to allow setting variables from 3 places with sensible precedence (CLI > env > defaults). See my comment below on whether we need this or not.